src/gt4py/testing/suites.py

```diff
@@ -467,19 +476,28 @@ def _test_generation(self, test, externals_dict):
 
         test["implementations"].append(implementation)
 
-    def _run_test_implementation(self, parameters_dict, implementation):
-
-        cls = type(self)
+    @classmethod
+    def _run_test_implementation(cls, parameters_dict, implementation):  # noqa: C901  # too complex
         input_data, exec_info = parameters_dict
 
         origin = cls.origin
         max_boundary = Boundary(cls.max_boundary)
+        field_params = cls.field_params
+        field_masks = {}
+        for name, value in input_data.items():
+            if isinstance(value, np.ndarray):
+                field_masks[name] = tuple(ax in field_params[name][0] for ax in ["I", "J", "K"])
```

Comment:
Minor: If you import `CartesianSpace` from `gt4py.definitions` above then this could be changed to
```
field_masks[name] = tuple(ax in field_params[name][0] for ax in CartesianSpace.names)
```

Reply:
Done